File: python/482_密钥格式化_easy.py
class Solution:
    def licenseKeyFormatting(self, s: str, k: int) -> str:
        ans, cnt = list(), 0
        for i in range(len(s)-1, -1, -1):
            if s[i] != '-':
                ans.append(s[i].upper())
                cnt += 1
                if cnt % k == 0:
                    ans.append('-')
        if ans and ans[-1] == '-':
            ans.pop()
        return ''.join(ans[::-1])

File: cpp/482_密钥格式化_easy.cpp
class Solution {
public:
    string licenseKeyFormatting(string s, int k) {
        string ans;
        int cnt = 0;
        for (int i=s.size()-1; i>=0; i--) {
            if (s[i] != '-') {
                ans.push_back(toupper(s[i]));
                cnt++;
                if (cnt % k == 0)
                    ans.push_back('-');
            }
        }
        if (ans.size() > 0 && ans.back() == '-')
            ans.pop_back();
        reverse(ans.begin(), ans.end());
        return ans;
    }
};